pandas/core/tools/timedeltas.py

```diff
@@ -106,26 +107,30 @@ def to_timedelta(arg, unit='ns', box=True, errors='raise'):
                                             box=box, errors=errors)
 
 
-def _coerce_scalar_to_timedelta_type(r, unit='ns', box=True, errors='raise'):
+def _coerce_scalar_to_timedelta_type(r, unit=None, box=True, errors='raise'):
     """Convert string 'r' to a timedelta object."""
 
     try:
-        result = convert_to_timedelta64(r, unit)
-    except ValueError:
-        if errors == 'raise':
-            raise
-        elif errors == 'ignore':
-            return r
+        result = parse_timedelta_string(r, unit)
+        result = np.timedelta64(result)
+    except (ValueError, TypeError):
+        try:
```

Comment:
why are you trying to convert this twice? if it fails aren't we done? (and either raise / ignore)

Reply:
This is supposed to be inline with what is being done in `array_to_timedelta64`:

https://github.com/pandas-dev/pandas/blob/6bf6cd2fe991f4f57e045e1f28a9dff860b72b37/pandas/_libs/tslibs/timedeltas.pyx#L235-L250

Although, come to think about it, there is somewhat duplicate logic and it might make sense to refactor `array_to_timedelta64` to use `_coerce_scalar_to_timedelta_type` in the for loop.

Reply:
@jbrockmendel just refactored all of this, and about about to merge. Let's wait till that lands.

Reply:
@jbrockmendel @jreback is it merged and can I proceed?

Reply:
you need to rebase on master